mbed/mbed_debug.py

```diff
@@ -0,0 +1,132 @@
+# mbed debug trace transformer, handling memory trace information
+# Author: Matthias L. Jugel (@thinkberg)
+#
+# Copyright (c) 2018 ubirch GmbH, All Rights Reserved
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied.
+
+import re
+import sys
+
+from serial.tools.miniterm import Transform
+
+def debug(s):
+    sys.stderr.write("D "+repr(s)+"\n")
+
+class MbedTransform(Transform):
+    mem = {}
+    allocated = 0
+    reset = None
+    buffer = ""
+
+    def __init__(self, reset_str=None):
+        self.reset = reset_str
+        if self.reset is not None:
+            sys.stderr.write("memory tracing enabled: resetting on '{}'\n".format(self.reset))
+
+        self.r_malloc = re.compile("([^#]*)#m:0x([0-9a-f]+);0x([0-9a-f]+)-(\\d+)")
+        self.r_calloc = re.compile("([^#]*)#c:0x([0-9a-f]+);0x([0-9a-f]+)-(\\d+);(\\d+)")
+        self.r_realloc = re.compile("([^#]*)#r:0x([0-9a-f]+);0x([0-9a-f]+)-0x([0-9a-f]+);(\\d+)")
+        self.r_free = re.compile("([^#]*)#f:0x([0-9a-f]+);0x([0-9a-f]+)-0x([0-9a-f]+)")
+
+    def rx(self, rx_input):
+        # collect lines
+        out = ""
+        for c in rx_input:
+            if c == '\r' or c == '\n':
+                if len(self.buffer):
+                    out += self.trace_line(self.buffer)
```

Comment:
This is the place where it runs the data through the trace matcher. The transformer class collects data until it sees either `\r` or `\n`

Reply:
Thanks for the reference